runtime/runc_jailer.go

```diff
@@ -41,38 +41,33 @@ const (
 
 // runcJailer uses runc to set up a jailed environment for the Firecracker VM.
 type runcJailer struct {
-	ctx    context.Context
-	logger *logrus.Entry
-	// ociBundlePath is the path that will be used to create an OCI bundle,
-	// https://github.com/opencontainers/runtime-spec/blob/master/bundle.md
-	ociBundlePath string
-	// runcBinaryPath is the path used to execute the runc binary from.
-	runcBinaryPath string
-	uid            uint32
-	gid            uint32
-	vmID           string
-	configSpec     specs.Spec
+	ctx        context.Context
+	logger     *logrus.Entry
+	Config     runcJailerConfig
+	vmID       string
+	configSpec specs.Spec
 }
 
 const firecrackerFileName = "firecracker"
 
-func newRuncJailer(
-	ctx context.Context,
-	logger *logrus.Entry,
-	vmID string,
-	ociBundlePath string,
-	runcBinaryPath string,
-	uid uint32,
-	gid uint32) (*runcJailer, error) {
-	l := logger.WithField("ociBundlePath", ociBundlePath).WithField("runcBinaryPath", runcBinaryPath)
+type runcJailerConfig struct {
```

Comment:
One general suggestion for the future is to consider making these changes (the introduction and use of the runcJailerConfig struct) in a separate commit. Still part of the same PR. Introduce the structure in an earlier commit, and refactor existing code to use it. Then add support for the cpuset fields in a second commit. It makes it very clear exactly why a given line of code is changing, and it makes it easier to debug and bisect the code if we ever need to.

Just something to think about for the future. Don't worry at all about changing what you've already done here.

Reply:
Damn, that's smart haha. Will definitely do that in the future.